tests/cpp/runtime/msgpack-test.cpp

```diff
@@ -0,0 +1,196 @@
+#include "runtime/kphp_core.h"
+#include "runtime/msgpack-serialization.h"
+#include <gtest/gtest.h>
+#include <utility>
+
+constexpr int64_t edge_int_values[] = {
+  0,
+  1,
+  -1,
+  std::numeric_limits<int64_t>::max(),
+  std::numeric_limits<int64_t>::min(),
+  std::numeric_limits<int32_t>::max(),
+  std::numeric_limits<int32_t>::min(),
+  std::numeric_limits<int32_t>::max() + 1_i64,
+  std::numeric_limits<int32_t>::min() + 1_i64,
+  std::numeric_limits<int32_t>::max() - 1_i64,
+  std::numeric_limits<int32_t>::min() - 1_i64,
+  std::numeric_limits<int16_t>::max(),
+  std::numeric_limits<int16_t>::min(),
+  std::numeric_limits<int16_t>::max() + 1_i64,
+  std::numeric_limits<int16_t>::min() + 1_i64,
+  std::numeric_limits<int16_t>::max() - 1_i64,
+  std::numeric_limits<int16_t>::min() - 1_i64,
+  std::numeric_limits<int8_t>::max(),
+  std::numeric_limits<int8_t>::min(),
+  std::numeric_limits<int8_t>::max() + 1_i64,
+  std::numeric_limits<int8_t>::min() + 1_i64,
+  std::numeric_limits<int8_t>::max() - 1_i64,
+  std::numeric_limits<int8_t>::min() - 1_i64,
+};
+constexpr double edge_fp_values[] = {+0, -0, 1.1e-300, -1.1e-300, 1, -1, std::numeric_limits<double>::min(), std::numeric_limits<double>::max()};
+
+const string edge_string_values[] = {
+  string(""),
+  string("0"),
+  string("a"),
+  string(" "),
+  string("   "),
+  string("\n"),
+  string("\r\n"),
+  string("\t"),
+  string("\a"),
+  string("\b"),
+  string("42num"),
+  string("num42"),
+  string("multi word"),
+  string("multi 12 word 23 with digits 42"),
+  string("! @ #  % ^ & * ()-_+="),
+  string("! @ #  % ^\0& * ()-_+="),
+};
+
+template<typename T>
+void assert_single(const T data) {
```

Comment:
It will be cool if you turn this assert in gtest test fixture. You can grep 'TEST_F' for more info here: https://github.com/google/googletest/blob/main/docs/primer.md

Reply:
I think it wouldn't improve the code. Firstly, we would lose type deducing in case of making `constexpr` arrays as non-static fields but we could leave them as `static constexpr` fields. Besides, `edge_string_fields` isn't `constexpr` so we also lose type deducing. I suppose this change will lead to inconsistency.

Reply:
ok, got it